homework-submission/React_HW2/todoApp2.jsx

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+// ContentDynamicTodo component is for map function to use
+const ContentDynamicTodo = ({ description, deadline, className, deleteItem }) => {
+  return (
+    <tr className={className} onClick={deleteItem}>
+      <td className="content"> {description} </td>
+      <td className="content"> {deadline} </td>
+      <td className="content"> {className} </td>
+    </tr>
+  );
+};
+
+// DynamicList where map function is
+const DynamicList = ({ list, deleteItem }) => {
+  const runTodoList = list.map(todo => (
+    <ContentDynamicTodo
+      key={todo.id}
+      description={todo.description}
+      deadline={todo.deadline}
+      className={todo.done ? '√' : 'x'}
+      deleteItem={() => {
+        deleteItem(todo.id);
+      }}
+    />
+  ));
+  return runTodoList;
+};
+
+// Here is our main Application component as a class
+class AppTodo2 extends React.Component {
+  //TodoList which contains new item as state
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoList: [
+        {
+          id: 1,
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          id: 2,
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+    };
+  }
+
+  newTodo = {
+    id: '',
+    description: 'Description',
+    deadline: 'No Deadline Entry',
+    done: false,
+  };
+
+  NewTodoDescriptionAndDeadline = () => {
+    return (
+      <div>
+        <div>
+          <label> Description </label>
+          <div id="containerDescriptionInput">
+            <input
+              type="text"
+              onChange={event => {
+                const inputValue = event.target.value;
+                this.setState({
+                  newTodo: {
+                    ...this.state.newTodo,
+                    description: inputValue,
+                    id: Math.random(),
```

Comment:
you still have the same mistake, I will try to be more clear.

an `onChange` event must never set the `id`, the place you want to set the `id` from is a place with no changes on every button, but just on button setting it and the next will generate a new one, what is happening here, is just as the following:
 - you are pressing a button on your keyboard,let's say you want to write 123
 - on each key you are pressing the id is changing here, and that means there will be a new id on each change you are trying to do.
 - the last key you have pressed, will stay the same until you press another one.

and that is the reason on the animated picture of the previous change, it haven't fixed yet try the following:
 - type something
 - then press `add`
 - change the date and press add again
 - try to remove one of them, both are removed

Reply:
I change that one let me check, I may PR the wrong file.